keras/layers/rnn/gru_test.py

```diff
@@ -993,6 +993,21 @@ def test_cloned_weight_names(self):
         clone_names = [x.name for x in clone.weights]
         self.assertEqual(model_names, clone_names)
 
+    def test_unique_state_names(self):
+        inp = keras.Input(batch_shape=[3, None, 3])
+        a = keras.layers.GRU(units=3, stateful=True)
+        b = keras.layers.GRU(units=3)  # no states
+        c = keras.layers.GRU(units=3, stateful=True)
+        _ = keras.Model(inp, [a(inp), b(inp), c(inp)])
+
+        names = set()
+
+        for l in [a, b, c]:
```

Comment:
For a unit test, let's put clear assertion in the body, and not have any looping/state merge.

How about:

self.assertEqual(a.states[0].name, '....')
self.assertEqual(b.states[0].name, '....')
....

Same for all other tests.